src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -198,8 +233,9 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             {
                 // Ensure that...
                 // 1. we move focus to the edit mode controls
-                // 2. this is the only entry that is in edit mode
-                for (uint32_t i = 0; i < _KeyBindingList.Size(); ++i)
+                // 2. any actions that were newly added are removed
+                // 3. this is the only entry that is in edit mode
+                for (int32_t i = _KeyBindingList.Size() - 1; i >= 0; --i)

```

Comment:
I understand reversing the loop here so you can `.RemoveAt` as you iterate... but why did this go to a signed integer from an unsigned one? I would think signed is undesirable here.

Reply:
Ah. Check out the last few iterations:
- `i=1` --> `i>=0` --> \<do stuff\> --> `--i`
- `i=0` --> `i>=0` --> \<do stuff\> --> `--i`
- (`uint32_t i`) `i=uintmax_t` --> `i>=0` --> cry 😭
- (`int32_t i`) `i=-1` --> `i>=0` is `false` --> exit loop

Reply:
and we can't change the condition to `i>0` because we still need it to run when `i=0` :(

I could add an `if (i==0) break;`, but I figured we'd never hit somebody that has more than `int32max_t` actions.

Reply:
Nnnnnnnnnn yeah ok fine.